addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -122,6 +123,44 @@ def _get_report_data(self, bom_id, searchQty=0, searchVariant=False):
             'precision': self.env['decimal.precision'].precision_get('Product Unit of Measure'),
         }
 
+    @api.model
+    def _get_components_closest_forecasted(self, lines, line_quantities, parent_bom, product_info, ignore_stock=False):
+        """
+            Returns a dict mapping a product to the closest date in the forecast report where
+            consummed quantity >= forecasted quantity.
+        """
+        if ignore_stock:
+            return {}
+        # Use defaultdict(list) in case there are lines with the same component.
+        closest_forecasted = defaultdict(list)
+        remaining_products = self.env['product.product']
+        product_quantities_info = {}
+        for line in lines:
+            product = line.product_id
+            quantities_info = product_quantities_info.setdefault(product.id, self._get_quantities_info(product, line.product_uom_id, parent_bom, product_info))
+            stock_loc = quantities_info['stock_loc']
+            product_info[product.id]['consumptions'][stock_loc] += line_quantities.get(line.id, 0.0)
+            if product.detailed_type != 'product':
+                # Use date.min as a sentinel value for _get_stock_availability
+                closest_forecasted[product.id].append(date.min)
+            elif quantities_info['stock_loc'] != 'in_stock':
+                closest_forecasted[product.id].append(date.max)
+            elif float_compare(product_info[product.id]['consumptions'][stock_loc], quantities_info['free_qty'], precision_rounding=product.uom_id.rounding) <= 0:
+                closest_forecasted[product.id].append(date.min)
+            else:
+                remaining_products |= product
+                closest_forecasted[product.id].append(date.max)
+        date_today = self.env.context.get('from_date', fields.date.today())
+        domain = [('state', '=', 'forecast'), ('date', '>=', date_today), ('product_id', 'in', remaining_products.ids)]
+        if self.env.context.get('warehouse'):
+            domain.append(('warehouse_id', '=', self.env.context.get('warehouse')))
+        res = self.env['report.stock.quantity']._read_group(domain, ['min_date:min(date)', 'product_id', 'product_qty'], ['product_id', 'product_qty'], lazy=False)
+        for group in res:
+            product_id = group['product_id'][0]
+            if group['product_qty'] >= product_info[product_id]['consumptions'][product_quantities_info[product_id]['stock_loc']]:
+                closest_forecasted[product_id].append(min(closest_forecasted[product_id], group['min_date']))
```

Comment:
By using `append()` after the _read_group(), we never clear the inital `date.max`. This removes the `Expected` dates from the report.
For reference, if a product isn't in enough quantity in stock, it's availability will be computed. If enough quantity is inbound (e.g. a PO due to be received), it will be marked as `Expected`. If not, then it will be maked as `Estimated` following the supplier delay.

Can I have your opinion on this?
```diff
            else:
                remaining_products |= product
-               closest_forecasted[product.id].append(date.max)
        date_today = self.env.context.get('from_date', fields.date.today())
        domain = [('state', '=', 'forecast'), ('date', '>=', date_today), ('product_id', 'in', remaining_products.ids)]
        if self.env.context.get('warehouse'):
            domain.append(('warehouse_id', '=', self.env.context.get('warehouse')))
        res = self.env['report.stock.quantity']._read_group(domain, ['min_date:min(date)', 'product_id', 'product_qty'], ['product_id', 'product_qty'], lazy=False)
        for group in res:
            product_id = group['product_id'][0]
            if group['product_qty'] >= product_info[product_id]['consumptions'][product_quantities_info[product_id]['stock_loc']]:
-               closest_forecasted[product_id].append(min(closest_forecasted[product_id], group['min_date']))
+               closest_forecasted[product_id].append(min(group['min_date'], date.max))
```
This solves the issue I mentionned, but it's rather untested at this point, so it may raise other issues.